nomos-mix/core/src/membership.rs

```diff
@@ -22,10 +23,20 @@ where
     M: MixMessage,
     M::PublicKey: PartialEq,
 {
-    pub fn new(mut nodes: Vec<Node<M::PublicKey>>, local_public_key: M::PublicKey) -> Self {
-        nodes.retain(|node| node.public_key != local_public_key);
+    pub fn new(nodes: Vec<Node<M::PublicKey>>, local_public_key: M::PublicKey) -> Self {
+        let mut remote_nodes = Vec::with_capacity(nodes.len() - 1);
+        let mut local_node = None;
+        nodes.into_iter().for_each(|node| {
+            if node.public_key == local_public_key {
+                local_node = Some(node);
+            } else {
+                remote_nodes.push(node);
+            }
+        });
+
```

Comment:
This could have probably be implemented using [`partition`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition) or [`extract_if`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extract_if)